### PR TITLE
Implement start time update via duration dropdown (mac)

### LINKED ISSUE #4591: Add ability to edit start time on duration dropdown
### 💻 Environment
Platform: macOS

### 📒 Description
User should be able to:
- Edit start time before starting new time entry
- Edit start time for running time entry
- Editing duration will update start time and other way around

### PR DESCRIPTION
**Pull request by AndrewVebster**
### 📒 Description

PR adds the ability to edit TE start time via duration dropdown

### 🕶️ Types of changes

- **New feature** (non-breaking change which adds functionality)

### 👫 Relationships

Closes #4591 

### 🔎 Review hints

Duration dropdown is hidden under local feature flag. To enable it go to `TimerDurationControl.swift` line 16 and set `isDurationDropdownEnabled` to `true`.

Test if interactions make sense and if using the start time and duration text fields work as expected.

### CODE REVIEW

**Review by skel35 — CHANGES_REQUESTED** [review]
The behavior is inconsistent with the web app.
STR
1. Timer is not running.
2. Set the start time to smth (e.g. `20:15`)

AR
The duration is not rounded down to minutes. E.g. `11:39 min`

ER
The duration is rounded down to minutes. E.g. `11:00 min`

File: src/toggl_api.cc [hunk 1]
```diff
@@ -642,6 +642,36 @@ char_t *toggl_format_tracked_time_duration(
...
+}
+
+int64_t toggl_timestamp_from_time_string(const char_t *time) {
+    int hours(0), minutes(0);
+    if (!toggl::Formatter::ParseTimeInput(to_string(time), &hours, &minutes)) {
+        return 0;
+    }
+
+    Poco::LocalDateTime now;
+    Poco::LocalDateTime local;
```
[the remark this answers is not in the document]
Comment (AndrewVebster) [reply src/toggl_api.cc]: updated 👍

**Comment by AndrewVebster** [comment]
I think on the Web app the duration is rounded because they also have an end time field. So after you enter start time the timer goes into "manual" mode and it's a little bit easier (duration = end - start).
Anyway, I did rounding to minutes. Maybe it's better to understand.

**Review by skel35 — APPROVED** [review]
Looks good.
I like both the behavior before the timer is started, and while the timer is running.

The web app's behavior is indeed different as it switches into the manual mode, so consistency is not so important. But having round minutes seems more clear and satisfying, so I like it more anyway. Feel free to ask more people about it if you think the previous behavior makes more sense.

**Pull request merged**